### PR TITLE
Refactor Camp and Payments for AuthDTO instead of token

### PR DESCRIPTION
**Pull request by danrye**
Refactoring both camps and payments here. Started with payments but camps was calling the controller for payments and needed updated as well.

### CODE REVIEW

File: Gateway/MinistryPlatform.Translation/Repositories/Interfaces/IContactRepository.cs [hunk 1]
```diff
@@ -16,7 +17,8 @@ public interface IContactRepository
         int GetContactIdByParticipantId(int participantId);
         List<MpHouseholdMember> GetHouseholdFamilyMembers(int householdId);
         List<MpHouseholdMember> GetOtherHouseholdMembers(int contactId);
-        MpMyContact GetMyProfile(string token);
+        MpMyContact GetMyProfile(AuthDTO token);
+        MpMyContact GetMyProfile(String token);
```
Comment (vchtelmakh) [inline Gateway/MinistryPlatform.Translation/Repositories/Interfaces/IContactRepository.cs]: Can be changed back to lowercase `string` for consistency, there is no functional difference.

File: Gateway/crds-angular/Services/CampService.cs [hunk 2]
```diff
@@ -133,7 +134,7 @@ public ProductDTO GetCampProductDetails(int eventId, int camperContactId, string
             return campProductInfo;
         }
 
-        public List<CampFamilyMember> GetEligibleFamilyMembers(int eventId, string token)
+        public List<CampFamilyMember> GetEligibleFamilyMembers(int eventId, AuthDTO token)
         {
             var apiToken = _apiUserRepository.GetDefaultApiClientToken();
             var myContact = _contactRepository.GetMyProfile(token);
```
Comment (andycanterbury) [inline Gateway/crds-angular/Services/CampService.cs]: The old token was used to get the profile, but only used the contact and household ids from that profile. Both of those ids are part of the AuthDTO object, so we can save a call to GetMyProfile here.

File: Gateway/crds-angular/Services/CampService.cs [hunk 3]
```diff
@@ -190,7 +191,7 @@ private CampFamilyMember NewCampFamilyMember(MpHouseholdMember member, int event
             };
         }
 
-        public void SaveCamperEmergencyContactInfo(List<CampEmergencyContactDTO> emergencyContacts, int eventId, int contactId, string token)
+        public void SaveCamperEmergencyContactInfo(List<CampEmergencyContactDTO> emergencyContacts, int eventId, int contactId, AuthDTO token)
         {
             var loggedInContact = _contactRepository.GetMyProfile(token);
```
Comment (andycanterbury) [inline Gateway/crds-angular/Services/CampService.cs]: Same here, the only thing used from this call is Household ID and we can get that from the AuthDTO object.

File: Gateway/crds-angular/Services/CampService.cs [hunk 4]
```diff
@@ -739,7 +740,7 @@ public MedicalInfoDTO GetCampMedicalInfo(int eventId, int contactId, string toke
             return camperMedInfo;
         }
 
-        public List<CampEmergencyContactDTO> GetCamperEmergencyContactInfo(int eventId, int contactId, string token)
+        public List<CampEmergencyContactDTO> GetCamperEmergencyContactInfo(int eventId, int contactId, AuthDTO token)
```
Comment (andycanterbury) [inline Gateway/crds-angular/Services/CampService.cs]: This method doesn't even use the token anywhere. We could just get rid of it.

File: Gateway/crds-angular/Services/CampService.cs [hunk 5]
```diff
@@ -270,7 +271,7 @@ public void SaveCamperEmergencyContactInfo(List<CampEmergencyContactDTO> emergen
             _formSubmissionRepository.SubmitFormResponse(formResponse);
         }
 
-        public CampReservationDTO SaveCampReservation(CampReservationDTO campReservation, int eventId, string token)
+        public CampReservationDTO SaveCampReservation(CampReservationDTO campReservation, int eventId, AuthDTO token)
```
Comment (andycanterbury) [inline Gateway/crds-angular/Services/CampService.cs]: Also token only used to get household Id

File: Gateway/crds-angular/Services/CampService.cs [hunk 6]
```diff
@@ -432,7 +433,7 @@ public void SetCamperAsRegistered(int eventId, int contactId)
             _eventRepository.SetParticipantAsRegistered(eventId, participant.ParticipantId);
         }
 
-        public List<MyCampDTO> GetMyCampInfo(string token)
+        public List<MyCampDTO> GetMyCampInfo(AuthDTO token)
         {
             var apiToken = _apiUserRepository.GetDefaultApiClientToken();
```
Comment (andycanterbury) [inline Gateway/crds-angular/Services/CampService.cs]: Another use of token just to get to household id

File: Gateway/crds-angular/Services/CampService.cs [hunk 7]
```diff
@@ -506,7 +507,7 @@ public void SaveWaivers(string token, int eventId, int contactId, List<CampWaive
             _eventRepository.SetWaivers(waiverResponses);
         }
 
-        public void SaveInvoice(CampProductDTO campProductDto, string token)
+        public void SaveInvoice(CampProductDTO campProductDto, AuthDTO token)
         {
             var loggedInContact = _contactRepository.GetMyProfile(token);
```
Comment (andycanterbury) [inline Gateway/crds-angular/Services/CampService.cs]: We did this a lot... all these can be updated.

File: Gateway/crds-angular/Services/CampService.cs [hunk 8]
```diff
@@ -604,7 +605,7 @@ public bool SendCampConfirmationEmail(int eventId, int invoiceId, int paymentId,
             }
         }
 
-        public void SaveCamperMedicalInfo(MedicalInfoDTO medicalInfo, int contactId, string token)
+        public void SaveCamperMedicalInfo(MedicalInfoDTO medicalInfo, int contactId, AuthDTO token)
```
Comment (andycanterbury) [inline Gateway/crds-angular/Services/CampService.cs]: Household Id again

File: Gateway/crds-angular/Services/CampService.cs [hunk 9]
```diff
@@ -675,7 +676,7 @@ public void SaveCamperMedicalInfo(MedicalInfoDTO medicalInfo, int contactId, str
 
 
 
-        public MedicalInfoDTO GetCampMedicalInfo(int eventId, int contactId, string token)
+        public MedicalInfoDTO GetCampMedicalInfo(int eventId, int contactId, AuthDTO token)
```
Comment (andycanterbury) [inline Gateway/crds-angular/Services/CampService.cs]: Household Id...

File: Gateway/crds-angular/Services/CampService.cs [hunk 10]
```diff
@@ -767,7 +768,7 @@ public List<CampEmergencyContactDTO> GetCamperEmergencyContactInfo(int eventId,
             return emergencyContacts;
         }
 
-        public CampReservationDTO GetCamperInfo(string token, int eventId, int contactId)
+        public CampReservationDTO GetCamperInfo(AuthDTO token, int eventId, int contactId)
```
Comment (andycanterbury) [inline Gateway/crds-angular/Services/CampService.cs]: And finally... household id.

File: Gateway/crds-angular/Services/PaymentService.cs [hunk 11]
```diff
@@ -319,15 +319,15 @@ public void UpdateInvoiceStatusAfterDecline(int invoiceId)
             _invoiceRepository.SetInvoiceStatus(invoiceId, paymentTotal > 0 ? _somepaidStatus : _nonePaidStatus);
         }
 
-        public bool DepositExists(int invoiceId, string token)
+        public bool DepositExists(int invoiceId, AuthDTO token)
         {
             var me = _contactRepository.GetMyProfile(token);
```
Comment (andycanterbury) [inline Gateway/crds-angular/Services/PaymentService.cs]: Only used to get Contact ID

File: Gateway/crds-angular/Services/PaymentService.cs [hunk 12]
```diff
@@ -363,7 +363,7 @@ public InvoiceDetailDTO GetInvoiceDetail(int invoiceId)
             return invoiceDetail;
         }
 
-        public void SendInvoicePaymentConfirmation(int paymentId, int invoiceId, string token)
+        public void SendInvoicePaymentConfirmation(int paymentId, int invoiceId, AuthDTO token)
         {
             var payment = _paymentRepository.GetPaymentById(paymentId);
             var me = _contactRepository.GetMyProfile(token);
```
Comment (andycanterbury) [inline Gateway/crds-angular/Services/PaymentService.cs]: Used to get contact_id and email address.

File: Gateway/MinistryPlatform.Translation/Repositories/ContactRepository.cs [hunk 13]
```diff
@@ -324,8 +325,13 @@ public List<MpHouseholdMember> GetOtherHouseholdMembers(int householdId)
                 Nickname = hm.Nickname,
             }).ToList();
             return householdMembers;
-        }      
+        }
+        public MpMyContact GetMyProfile(AuthDTO token)
+        {
+            return this.GetContactById(token.UserInfo.Mp.ContactId);
```
Comment (andycanterbury) [inline Gateway/MinistryPlatform.Translation/Repositories/ContactRepository.cs]: Somebody has been writing too much JS. 😛 `this.` isn't needed here.

**Pull request merged**